### PR TITLE
core: add is_headless to specify hidden window before init

### PR DESCRIPTION
**Pull request by undefinedDarkness**
I hope it's no trouble 

- [X] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

**Comment by slimsag** [comment]
Thanks, I'm happy to accept an improvement like this in general but I'm afraid this is not the right API.

Can you describe the issue you want to solve here? Is it:

a. You don't want to have to call `setHeadless` (API is burdensome)?
b. You are seeing the window appear and then disappear (flicker), and need a way to specify headless earlier?

The reason this enum-based API is not great is because one could easily have e.g. a game application, which sometimes may be run `game.exe --headless` - so in that case what should the user pick?

File: libs/core/src/Core.zig [hunk 1]
```diff
@@ -7,8 +7,10 @@ pub const Core = @This();
 
 internal: platform.Core,
 
+pub const AppType = enum { Standard, Game, Headless };
+
 pub const Options = struct {
-    is_app: bool = false,
+    app_type: AppType = .Game,
```
Comment (slimsag) [inline libs/core/src/Core.zig:13]: I suspect the solution here will be to:

1. Keep `is_app: bool = false,`
2. Additionally add `headless: bool = false,` (below `size` field) which we can respect at `init`

Comment (undefinedDarkness) [reply libs/core/src/Core.zig:13]: Sure, that works as well

Comment (slimsag) [reply libs/core/src/Core.zig:13]: cool, if you're happy with that I'm happy

want to send a new PR or update this one? or prefer if I make that change? either way is fine with me

**Comment by undefinedDarkness** [comment]
> You are seeing the window appear and then disappear (flicker), and need a way to specify headless earlier?
Yeah just this bit, the rest doesn't matter, The window stays open till core initiates which takes a few,

File: libs/core/src/Core.zig [hunk 2]
```diff
@@ -8,7 +8,8 @@ pub const Core = @This();
 internal: platform.Core,
 
 pub const Options = struct {
-    is_app: bool = false,
+    is_app: bool = true,
```
Comment (slimsag) [inline libs/core/src/Core.zig:11]:
```suggestion
    is_app: bool = false,
```

**Review by slimsag — APPROVED** [review]

**Comment by slimsag** [comment]
Thanks!

**Pull request merged**